src/mg/PAMI/Message/IncomingMessage.php

```diff
@@ -81,6 +87,33 @@ public function getRawContent()
     }
 
     /**
+     * Returns the channel variables for all reported channels.
+     * https://github.com/marcelog/PAMI/issues/85
+     *
+     * The channel names will be lowercased.
+     *
+     * @return array
+     */
+    public function getAllChannelVariables()
+    {
+      return $this->channelVariables;
+    }
+
+    /**
+     * Returns the channel variables for the given channel.
+     * https://github.com/marcelog/PAMI/issues/85
+     *
+     * @param string $channel Channel name. If not given, will return variables
+     * for the "current" channel.
+     *
+     * @return array
+     */
+    public function getChannelVariables($channel = 'default')
```

Comment:
@jacobkiers I've added support for strange use cases "just in case":
 * If no "Channel" key is present but "ChannelVariable" is there, will save it as channel "default".
 * If no "Channel" key is present but "ChannelVariable(xxx)" keys are present, each variable will be saved under that channel name.
* If "Channel" key is present and "ChannelVariable(xxx)" keys are present, each variable will be saved under that channel name. The channel variables for the channel specified by "Channel" and "ChannelVariable(xxx)" will be merged when they match.